### DISCUSSION TITLE
PolarFire SoC SPI Flash example (external Micron flash)

### DISCUSSION THREAD

**Discussion by bilievski** [polarfire-soc/polarfire-soc-documentation#334] [Bare metal embedded software]
The SPI flash example project for writing to an external Micron flash does not enable the clock to the SPI_0 MSS peripheral. The current code only de-asserts the reset bit of the peripheral on line 58 of u54_1.c.

Turning on the SPI_0 MSS peripheral required me to add the following to the u54 application code:
`    (void)mss_config_clk_rst(MSS_PERIPH_SPI0, (uint8_t) MPFS_HAL_FIRST_HART, PERIPHERAL_ON);
`

I'm not sure if this was intended, but I've perused the other SPI loopback example project and it calls the above function to get both SPI peripherals up and running.

**Comment by nitindeshpande** [comment]
@bilievski 
You are correct. You do need a call to mss_config_clk_rst(). Were you able to get the SPI master to communicate with a Flash memory? which one?

About the 'mpfs-spi-flash' project, as the readme suggest, this is a reference project and not tested on icicle kit. We are in process of updating this example project with actual SPI flash tested on the icicle kit. We will have this published soon.
